taichi/codegen/llvm/codegen_llvm.cpp

```diff
@@ -2783,6 +2788,48 @@ void TaskCodeGenLLVM::set_struct_to_buffer(
                        current_element, current_index);
 }
 
+llvm::Value *TaskCodeGenLLVM::get_struct_arg(std::vector<int> index) {
+  auto *args_ptr = get_args_ptr(current_callable, get_context());
+  auto *args_type = current_callable->args_type;
+  auto *arg_type = args_type->get_element_type(index);
+  std::vector<llvm::Value *> gep_index;
```

Comment:
nit:
```suggestion
  std::vector<llvm::Value *> gep_index(index.size() + 1);
```
Then index into it